pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter.go

```diff
@@ -59,6 +60,11 @@ const (
 	maxDNSSearchListChars = 256
 )
 
+const (
+	multiQueueMinQueues = 2
```

Comment:
Not sure we want a lower limit. If I only have one vCPU and it only makes sense to make then one queue, I would be fine with only getting one queue.

Reply:
This might be confusing for a user who requested multiple queues to get only one queue.
But I agree that we don't need a const for this.

Reply:
> This might be confusing for a user who requested multiple queues to get only one queue.

Maybe. I see potential for both to be confusing. I guess it depends on why people enable the flag and how much they know about internals. In this case I would vote or keeping it, since it is not "wrong" and means an API breakage. The upper limit makes sense I guess (would we see an error from the kernel or so?).

Reply:
My point is that is you request multi-queue, you should get multi-queue. 

Now, if I've understood that flag wrong and it actually means "create as many queues as kubevirt sees fit", this approach is wrong, and we should instead not create the tap device with multiple queues whenever we get a request w/ 1 vCPU .

Which one is it ? 

Currently the [API says](https://github.com/kubevirt/kubevirt/blob/230a250b834fb68b38669185f772104cb34a9928/staging/src/kubevirt.io/client-go/api/v1/schema.go#L372) "If specified, virtual network interfaces configured with a virtio bus will also enable the vhost multiqueue feature"; it might be me, but that misleads me into believing KubeVirt will configure me the interfaces in multi-queue mode.

Reply:
Something simple like this may help:

"If specified, virtual network interfaces configured with a virtio bus will also enable the vhost multiqueue feature for block devices. The number of queues created depends on additional factors of the VirtualMachineInstance, like the number of guest CPUs."

Reply:
Another side question: In case we start CPU hot-plug at some point, would there dynamically new queues created?

Reply:
> > * change the scope of this PR to only add the upper boundary on the number of queues
> 
> Thinking about it, this is just the upper limit of queues, right? 

Correct, but I think it has that number to match the max number of vCPUs in guest, as per [this comment](https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/drivers/net/tun.c?id=refs/tags/v4.1.26#n126):

```
/* MAX_TAP_QUEUES 256 is chosen to allow rx/tx queues to be equal
 * to max number of VCPUs in guest. */
#define MAX_TAP_QUEUES 256
...
```

> Not of CPUs. Can't we just only create at max 256 queues? I am not sure if libvirt creates so many queues. We tell libvirt the number of queues to create ,right? So we could just not create more than 256 at max. @vladikr ?
> 
> If so, I would not do a validation and really just create the appropriate amount of queues and open the tap device the right way.